tools/msms_extractor/msms_extractor.xml

```diff
@@ -0,0 +1,84 @@
+
+<tool id="msms_extractor" name="msms_extractor" version="1.0.0">
+  <description>Extract MS/MS scans from the mzML file(s) based on PSM report.</description>
+  <requirements>
+      <requirement type="package" version="3_0_9992">proteowizard</requirement>
+      <requirement type="package" version="3.4">pyteomics</requirement>
+  </requirements>
+  <command detect_errors="exit_code"><![CDATA[python msms_extractor.py $spectrumfile $psmreportfile $output ${spectrumfile.name.rsplit('.',1)[0]} $removeretain.doremoveretain $removeretain.num_random_scans]]></command>
```

Comment:
please use `$__tool_directory__/msms_extractor.py` and single-quote all text and data-parameters.